### PR TITLE
CubeViz infrastructure refactor

### PR DESCRIPTION
**Pull request by nmearl**
Infrastructure refactor for use as a Glue plugin.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -103,21 +102,23 @@ matrix:
...
+        # Do a PEP8 test with pycodestyle
+        - os: linux
+          env: MAIN_CMD='pycodestyle packagename --count' SETUP_CMD=''
 
     allow_failures:
         # Do a PEP8 test with pycodestyle
         # (allow to fail unless your code completely compliant)
         - os: linux
-          env: MAIN_CMD='pycodestyle specviz --count' SETUP_CMD=''
+          env: MAIN_CMD='pycodestyle packagename --count' SETUP_CMD=''
```
Comment (astrofrog) [inline .travis.yml]: Here and above, looks like this needs to be changed back to specviz

File: .travis.yml [hunk 2]
```diff
@@ -150,4 +151,4 @@ script:
 after_success:
     # If coveralls.io is set up for this package, uncomment the line below.
     # The coveragerc file may be customized as needed for your package.
-    # - if [[ $SETUP_CMD == *coverage* ]]; then coveralls --rcfile='specviz/tests/coveragerc'; fi
+    # - if [[ $SETUP_CMD == *coverage* ]]; then coveralls --rcfile='packagename/tests/coveragerc'; fi
```
Comment (astrofrog) [inline .travis.yml]: Same here

File: setup.cfg [hunk 3]
```diff
@@ -47,7 +47,7 @@ url = https://specviz.readthedocs.io
 edit_on_github = True
 github_project = spacetelescope/specviz
 # version should be PEP386 compatible (http://www.python.org/dev/peps/pep-0386)
-version = 0.4.2
+version = 0.4.1
```
Comment (astrofrog) [inline setup.cfg]: Is this correct?

File: specviz/third_party/glue/data_viewer.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    # to updates to data or subsets.
+
+    def _update_data(self, message):
+        print("Updating data")
+
+    def _add_subset(self, message):
+        self.add_subset(message.subset)
+
+    def _update_subset(self, message):
+        print("Updating subset")
```
Comment (astrofrog) [inline specviz/third_party/glue/data_viewer.py]: ``_update_data``, ``_update_subset``, and ``_remove_subset`` should probably be implemented before merging?

File: .travis.yml (hunk 1 above)
Comment (nmearl) [reply .travis.yml]: Oops, good catch.

**Comment by nmearl** [comment]
@astrofrog Give this version a try; includes propagation of subset selections.

**Pull request merged**